src/wallet/wallet.h

```diff
@@ -204,6 +204,7 @@ class CAddressBookData
     std::string m_label;
 public:
     std::string purpose;
+    bool m_used{false};
```

Comment:
It's indeed confusing that there is an existing "used" that doesn't actually indicate usedness.

But this field *is* actually the correct value, so I'm not sure what else it might be named?

Reply:
Added the doc

Reply:
Can we just reuse existing `StringMap destdata` "used" key to store the correct "used" state?

Reply:
I don't know if there's usage of the incorrect "used" flag.

Also, I don't think it makes sense to store this one.

Reply:
Based on my research the existing flag is used for "avoid reuse" feature in the wallet. Sounds like a good idea to fix it.